sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -73,6 +73,84 @@ class CollationStringExpressionsSuite extends QueryTest
     })
   }
 
+  test("SUBSTRING_INDEX check result on explicitly collated strings") {
+    def testSubstringIndex(str: String, delim: String, cnt: Integer,
+                           collationId: Integer, expected: String): Unit = {
+      val string = Literal.create(str, StringType(collationId))
+      val delimiter = Literal.create(delim, StringType(collationId))
+      val count = Literal(cnt)
+
+      checkEvaluation(SubstringIndex(string, delimiter, count), expected)
+    }
+
+    testSubstringIndex("wwwgapachegorg", "g", -3, 0, "apachegorg")
+    testSubstringIndex("www||apache||org", "||", 2, 0, "www||apache")
+    // UTF8_BINARY_LCASE
```

Comment:
instead of separating these blocks with comments describing the collation used how about using a variable denoting the current collation
```
var collationId = CollationFactory.nameToId("UTF8_BINARY")
testcases ...

collationId = CollationFactory.nameToId("UNICODE")
testcases ...

```

this way we won't have to litter the code with 'magic constants' which should make it more readable

Reply:
also we shouldn't treat these values as never changing and should always refer to collations by their name rather than their id